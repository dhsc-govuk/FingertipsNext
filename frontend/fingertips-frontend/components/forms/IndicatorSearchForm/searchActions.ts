import { SearchParams, SearchStateManager } from '@/lib/searchStateManager';
import { redirect, RedirectType } from 'next/navigation';
import { z } from 'zod';

const $IndicatorSearchFormSchema = z.object({
  indicator: z
    .string()
    .trim()
    .min(1, { message: 'Please enter an indicator id' }),
});

export type State = {
  errors?: {
    indicator?: string[];
  };
  message?: string | null;
};

export type IndicatorSearchForm = z.infer<typeof $IndicatorSearchFormSchema>;

export type IndicatorSearchFormState<T = IndicatorSearchForm> = State & T;

export async function searchIndicator(
  prevState: IndicatorSearchFormState,
  formData: FormData
): Promise<IndicatorSearchFormState> {
  const validatedField = $IndicatorSearchFormSchema.safeParse({
    indicator: formData.get('indicator'),
  });

  if (!validatedField.success) {
    return {
      indicator: formData.get('indicator')?.toString().trim() ?? '',
      errors: validatedField.error.flatten().fieldErrors,
      message: 'Please enter a subject',
    };
  }

  const { indicator } = validatedField.data;

  const searchState = new SearchStateManager({
    [SearchParams.SearchedIndicator]: indicator,
  });
  redirect(searchState.generatePath('results'), RedirectType.push);
}
