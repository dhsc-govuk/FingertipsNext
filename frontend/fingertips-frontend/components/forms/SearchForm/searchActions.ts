'use server';

import { z } from 'zod';
import { redirect, RedirectType } from 'next/navigation';
import { SearchStateManager } from '@/lib/searchStateManager';

const $SearchFormSchema = z.object({
  indicator: z
    .string()
    .trim()
    .min(1, { message: 'Please enter an indicator id' }),
});

export type State = {
  errors?: {
    indicator?: string[];
  };
  message?: string | null;
};

export type SearchForm = z.infer<typeof $SearchFormSchema>;

export type SearchFormState<T = SearchForm> = State & T;

export async function searchIndicator(
  prevState: SearchFormState,
  formData: FormData
): Promise<SearchFormState> {
  const validatedFields = $SearchFormSchema.safeParse({
    indicator: formData.get('indicator'),
  });

  if (!validatedFields.success) {
    return {
      indicator: formData.get('indicator')?.toString().trim() ?? '',
      errors: validatedFields.error.flatten().fieldErrors,
      message: 'Please enter a value for the indicator field',
    };
  }

  const { indicator } = validatedFields.data;

  const searchState = new SearchStateManager(indicator);
  redirect(searchState.generatePath('/search/results'), RedirectType.push);
}
