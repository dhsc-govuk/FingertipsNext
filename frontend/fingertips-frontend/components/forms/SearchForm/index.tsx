'use client';

import { Button, InputField, H3 } from 'govuk-react';
import { spacing } from '@govuk-react/lib';
import styled from 'styled-components';
import { GovukColours } from '@/lib/styleHelpers/colours';
import { AreaAutoCompleteInputField } from '@/components/molecules/AreaAutoCompleteInputField';
import { SearchParams, SearchStateParams } from '@/lib/searchStateManager';
import { SearchFormState } from '@/components/forms/SearchForm/searchActions';
import { SelectedAreasPanel } from '@/components/molecules/SelectedAreasPanel';
import { AreaWithRelations } from '@/generated-sources/ft-api-client';
import {
  AreaFilterData,
  SelectAreasFilterPanel,
} from '@/components/molecules/SelectAreasFilterPanel';
import { ShowHideContainer } from '@/components/molecules/ShowHideContainer';
import { ALL_AREAS_SELECTED } from '@/lib/areaFilterHelpers/constants';

const StyledInputField = styled(InputField)(
  spacing.withWhiteSpace({ marginBottom: 6 })
);

interface SearchFormProps {
  searchState?: SearchStateParams;
  formState: SearchFormState;
  selectedAreasData?: AreaWithRelations[];
  areaFilterData?: AreaFilterData;
}

export const SearchForm = ({
  searchState,
  formState,
  selectedAreasData,
  areaFilterData,
}: Readonly<SearchFormProps>) => {
  const selectedAreas = searchState?.[SearchParams.AreasSelected];

  return (
    <div data-testid="search-form">
      <H3>Find public health data</H3>
      <input
        name="searchState"
        defaultValue={JSON.stringify(searchState)}
        hidden
      />
      <StyledInputField
        input={{
          id: 'indicator',
          name: 'indicator',
          defaultValue: formState.indicator ?? '',
        }}
        hint={
          <div style={{ color: GovukColours.DarkGrey }}>
            For example, smoking, diabetes prevalence, or a specific indicator
            ID
          </div>
        }
        meta={{
          touched: !!formState.message,
          error: 'Enter a subject you want to search for',
        }}
        data-testid="indicator-search-form-input"
      >
        Search by subject
      </StyledInputField>
      <AreaAutoCompleteInputField
        key={`area-auto-complete-${JSON.stringify(searchState)}`}
        inputFieldErrorStatus={!!formState.message}
        searchState={searchState}
        firstSelectedArea={selectedAreasData?.[0]}
      />

      {(selectedAreas && selectedAreas.length > 0) ||
      searchState?.[SearchParams.GroupAreaSelected] === ALL_AREAS_SELECTED ? (
        <SelectedAreasPanel
          key={`selected-area-panel-${JSON.stringify(searchState)}`}
          searchState={searchState}
          areaFilterData={areaFilterData}
          selectedAreasData={selectedAreasData}
          inFilterPane={false}
        />
      ) : null}

      <ShowHideContainer summary="Filter by area" open={false}>
        <SelectAreasFilterPanel
          key={`area-filter-panel-${JSON.stringify(searchState)}`}
          areaFilterData={areaFilterData}
          searchState={searchState}
        />
      </ShowHideContainer>

      <Button
        type="submit"
        data-testid="search-form-button-submit"
        style={{ marginTop: '25px' }}
      >
        Search
      </Button>
    </div>
  );
};
