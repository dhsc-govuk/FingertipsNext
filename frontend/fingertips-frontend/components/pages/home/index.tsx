'use client';

import {
  ErrorSummary,
  H2,
  H3,
  InsetText,
  Link,
  ListItem,
  Paragraph,
  SectionBreak,
  UnorderedList,
} from 'govuk-react';
import { SearchForm } from '@/components/forms/SearchForm';
import {
  SearchFormState,
  searchIndicator,
} from '@/components/forms/SearchForm/searchActions';
import { useActionState, useEffect } from 'react';
import styled from 'styled-components';
import { spacing } from '@govuk-react/lib';
import { Area } from '@/generated-sources/ft-api-client';
import { AreaFilterData } from '@/components/molecules/SelectAreasFilterPanel';
import { siteTitle } from '@/lib/constants';

export const ZeroMarginParagraph = styled(Paragraph)(
  spacing.withWhiteSpace({ marginBottom: 0 })
);

interface HomeProps {
  areaFilterData?: AreaFilterData;
  selectedAreasData?: Area[];
  initialFormState: SearchFormState;
}
export const Home = ({
  areaFilterData,
  initialFormState,
  selectedAreasData,
}: HomeProps) => {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const [formState, setFormState] = useActionState(
    searchIndicator,
    initialFormState
  );

  return (
    <div>
      {formState.message && (
        <ErrorSummary
          errors={[
            {
              targetName: 'indicator',
              text: 'Enter a subject you want to search for',
            },
            {
              targetName: 'areaSearched',
              text: 'Enter an area you want to search for',
            },
          ]}
          data-testid="search-form-error-summary"
          onHandleErrorClick={(targetName: string) => {
            const targetElement = document.getElementById(targetName);
            targetElement?.scrollIntoView(true);
            targetElement?.focus();
          }}
        />
      )}
      <H2>{siteTitle}</H2>
      <Paragraph>
        This service provides easy access to a comprehensive collection of
        public health data for England. It can be used to monitor trends,
        identify inequalities, and make make data-driven decisions to improve
        public health.
      </Paragraph>

      <InsetText>This service was previously known as Fingertips</InsetText>

      <ZeroMarginParagraph>Contents</ZeroMarginParagraph>
      <UnorderedList listStyleType='"â€” "'>
        <ListItem>
          <Link href="#search">Find public health data</Link>
        </ListItem>
        <ListItem>
          <Link href="#whatfor">What the service is for</Link>
        </ListItem>
        <ListItem>
          <Link href="#indicators">About indicators</Link>
        </ListItem>
        <ListItem>
          <Link href="#whofor">Who the service is for</Link>
        </ListItem>
      </UnorderedList>

      <div id="search">
        <form action={setFormState}>
          <SearchForm
            formState={formState}
            selectedAreasData={selectedAreasData}
            areaFilterData={areaFilterData}
          />
        </form>
      </div>
      <SectionBreak level="LARGE" visible />
      <H3 id="whatfor">What the service is for</H3>
      <Paragraph>You can use this service to:</Paragraph>
      <UnorderedList>
        <ListItem>
          Identify and compare health trends and inequalities using data
          visualisations
        </ListItem>
        <ListItem>Download datasets and visualisations</ListItem>
        <ListItem>
          Access raw data via an application programming interface (API)
        </ListItem>
      </UnorderedList>
      <Paragraph>
        This data comes from trusted sources such as the NHS and Office of
        National Statistics (ONS). It is analysed to create a wide range of
        health and wellbeing data sets known as indicators, and themed topics
        known as profiles.
      </Paragraph>
      <br />
      <H3 id="indicators">About indicators</H3>
      <Paragraph>
        Indicators use data to show us how things are going and if there are any
        changes over time. For example, if the number of people smoking in South
        West England has gone up or down in the last ten years. These indicators
        can be used to compare public heath by areas, and see how it is impacted
        by inequalities such as age, sex or ethnicity.
      </Paragraph>
      <br />
      <H3 id="whofor">Who the service is for</H3>
      <Paragraph>
        The service is free and available to everyone. Its primary role is to
        support health professionals, policymakers, and researchers to monitor
        health trends, identify inequalities, and make informed decisions about
        public health in England.
      </Paragraph>
    </div>
  );
};
