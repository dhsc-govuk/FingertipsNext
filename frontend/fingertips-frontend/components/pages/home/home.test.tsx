import { render, screen /*, waitFor, within */ } from '@testing-library/react';
import { Home } from '@/components/pages/home/index';
import { SearchFormState } from '@/components/forms/SearchForm/searchActions';
import { expect } from '@jest/globals';
// import { userEvent } from '@testing-library/user-event';

const initialState: SearchFormState = {
  indicator: '',
  areaSearched: '',
  message: null,
  errors: {},
};

it('snapshot test - renders the homepage', () => {
  const container = render(<Home searchFormState={initialState} />);

  expect(container.asFragment()).toMatchSnapshot();
});

it('should render an indicator link', () => {
  render(<Home searchFormState={initialState} />);
  const link = screen.getByRole('link', { name: /indicators/ });

  expect(link).toBeInTheDocument();
});

it('should render an indicator link that points to the indicator section', () => {
  render(<Home searchFormState={initialState} />);
  const link = screen.getByRole('link', { name: /indicators/ });

  expect(link).toHaveAttribute('href', '#indicators');
});

it('should render the SearchForm component', () => {
  render(<Home searchFormState={initialState} />);
  const searchForm = screen.getByTestId('search-form');

  expect(searchForm).toBeInTheDocument();
});

it('should display the error summary component when there is a validation error', () => {
  const errorState: SearchFormState = {
    indicator: '',
    areaSearched: '',
    message: 'Error message',
    errors: {},
  };

  render(<Home searchFormState={errorState} />);

  expect(screen.getByTestId('search-form-error-summary')).toBeInTheDocument();
});

// it('should display the error summary component when there is a validation error', async () => {
//   // Add missing function to jsdom
//   const scrollMock = jest.fn();
//   window.HTMLElement.prototype.scrollIntoView = scrollMock;
//
//   const user = userEvent.setup();
//
//   const errorState: SearchFormState = {
//     indicator: '',
//     areaSearched: '',
//     message: 'Error message',
//     errors: {},
//   };
//
//   render(<Home searchFormState={errorState} />);
//
// const searchForm = screen.getByTestId('search-form')
//   const anchor = within(searchForm).getByText('Indicator field').closest('a');
//   if (anchor) {
//     await user.click(anchor);
//   }
//
//   await waitFor(() => {
//     expect(screen.getByRole('textbox', { name: /indicator/i })).toHaveFocus();
//   });
//   expect(scrollMock).toBeCalledTimes(1);
// });
