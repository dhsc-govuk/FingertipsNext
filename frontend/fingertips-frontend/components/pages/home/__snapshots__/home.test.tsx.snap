// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test - renders the homepage 1`] = `
<DocumentFragment>
  .c27 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 19px;
  box-sizing: border-box;
  display: inline-block;
  position: relative;
  width: 100%;
  margin-top: 0;
  padding: 7px 10px;
  border: 2px solid transparent;
  border-radius: 0;
  color: #ffffff;
  background-color: #00703c;
  box-shadow: 0 2px 0 #002413;
  text-align: center;
  vertical-align: top;
  cursor: pointer;
  -webkit-appearance: none;
  margin-bottom: 22px;
}

.c27:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c27:link,
.c27:visited,
.c27:active,
.c27:hover {
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c27::-moz-focus-inner {
  padding: 0;
  border: 0;
}

.c27:hover,
.c27:focus {
  background-color: #00572e;
}

.c27:active {
  top: 2px;
  box-shadow: none;
}

.c27::before {
  content: "";
  display: block;
  position: absolute;
  top: -2px;
  right: -2px;
  bottom: -4px;
  left: -2px;
  background: transparent;
}

.c27:active::before {
  top: -4px;
}

.c27:disabled {
  opacity: 0.5;
  background: #00703c;
}

.c27:disabled:hover {
  background-color: #00703c;
  cursor: default;
}

.c27:disabled:focus {
  outline: none;
}

.c27:disabled:active {
  top: 0;
  box-shadow: 0 2px 0 #002413;
}

.c27 svg {
  max-width: 15px;
}

.c12 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  display: block;
  margin-bottom: 15px;
  color: #6f777b;
}

.c11 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c13 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 40px;
  margin-top: 0;
  padding: 5px;
  border: 2px solid #0b0c0c;
  border-radius: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin-bottom: 0;
}

.c13:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c13::-webkit-outer-spin-button,
.c13::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
}

.c13[type="number"] {
  -moz-appearance: textfield;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c9:after {
  content: '';
  display: table;
  clear: both;
}

.c16 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-bottom: 20px;
  display: block;
}

.c19 {
  display: inline-block;
  position: relative;
  margin-bottom: 5px;
  padding-left: 25px;
  color: #1d70b8;
  cursor: pointer;
}

.c19:hover {
  color: #2b8cc4;
}

.c19:focus {
  outline: 4px solid #ffdd00;
  outline-offset: -1px;
  color: #0b0c0c;
  background: #ffdd00;
}

.c19::-webkit-details-marker {
  display: none;
}

.c19:before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,100% 50%,0% 100%);
  clip-path: polygon(0% 0%,100% 50%,0% 100%);
  border-width: 7px 0 7px 12.124px;
  border-left-color: inherit;
}

[open] > .c18:before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,50% 100%,100% 0%);
  clip-path: polygon(0% 0%,50% 100%,100% 0%);
  border-width: 12.124px 7px 0 7px;
  border-top-color: inherit;
}

.c20 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c21 {
  padding: 15px;
  padding-left: 20px;
  border-left: 5px solid #bfc1c3;
}

.c21 p {
  margin-top: 0;
  margin-bottom: 20px;
}

.c21 > :last-child,
.c21 p:last-child {
  margin-bottom: 0;
}

.c6 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 5px;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  list-style-type: "â€” ";
  padding-left: 20px;
}

.c3 .c3 {
  margin-top: 10px;
}

.c4 > .c5 {
  margin-bottom: 5px;
}

.c29 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-top: 0;
  margin-bottom: 15px;
  padding-left: 20px;
}

.c3 .c3 {
  margin-top: 10px;
}

.c29 > .c5 {
  margin-bottom: 5px;
}

.c0 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 24px;
  line-height: 1.0416666666666667;
  display: block;
  margin-top: 0;
  margin-bottom: 20px;
}

.c8 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c7 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c7:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
  background-color: #ffdd00;
}

.c7:link {
  color: #1d70b8;
}

.c7:visited {
  color: #4c2c92;
}

.c7:hover {
  color: #2b8cc4;
}

.c7:active {
  color: #2b8cc4;
}

.c7:focus {
  color: #0b0c0c;
}

.c1 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 15px;
}

.c1 > p {
  margin: 0;
}

.c1 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c1 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c1 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

.c25 {
  margin-bottom: 20px;
}

.c24 .c24:last-of-type {
  margin-bottom: 0;
}

.c28 {
  margin: 0;
  border: 0;
  margin-top: 20px;
  margin-bottom: 20px;
  border-bottom: 1px solid #bfc1c3;
}

.c23 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c23:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c15 {
  margin-bottom: 5px;
}

.c14 {
  margin-bottom: 1em;
}

.c22 {
  margin-bottom: 1em;
}

.c22 span {
  font-weight: bold;
}

.c22 select {
  width: 100%;
}

.c26 {
  font-weight: bold;
}

.c17 div {
  padding: 0.5em 0em;
}

.c17 summary {
  color: #000000;
}

.c17 summary:hover {
  color: #000000;
}

.c17 div {
  border-left: none;
}

.c10 {
  margin-bottom: 20px;
}

.c2 {
  margin-bottom: 0;
}

@media print {
  .c27 {
    font-size: 14px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c27 {
    font-size: 19px;
    line-height: 19px;
  }
}

@media only screen and (min-width:641px) {
  .c27 {
    width: auto;
  }
}

@media only screen and (min-width:641px) {
  .c27 svg {
    margin-left: 10px;
  }
}

@media only screen and (min-width:641px) {
  .c27 {
    margin-bottom: 32px;
  }
}

@media print {
  .c12 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c12 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c11 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c11 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c11 {
    margin-bottom: 0;
  }
}

@media print {
  .c13 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c13 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c13 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c9 {
    margin-bottom: 0;
  }
}

@media print {
  .c16 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c16 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c16 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c16 {
    margin-bottom: 30px;
  }
}

@media print {
  .c6 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c6 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c4 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    margin-bottom: 20px;
  }
}

@media print {
  .c29 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c29 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c29 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c29 {
    margin-bottom: 20px;
  }
}

@media print {
  .c0 {
    color: #000;
  }
}

@media print {
  .c0 {
    font-size: 24px;
    line-height: 1.05;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 36px;
    line-height: 1.1111111111111112;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

@media print {
  .c8 {
    color: #000;
  }
}

@media print {
  .c8 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c8 {
    margin-bottom: 20px;
  }
}

@media print {
  .c7 {
    font-family: sans-serif;
  }
}

@media print {
  .c7[href^="/"]::after,
  .c7[href^="http://"]::after,
  .c7[href^="https://"]::after {
    content: " (" attr(href) ")";
    font-size: 90%;
    word-wrap: break-word;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

@media only screen and (min-width:641px) {
  .c25 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c28 {
    margin-top: 30px;
    margin-bottom: 30px;
  }
}

@media print {
  .c23 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c23 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c23 {
    width: 50%;
    height: 38px;
  }
}

@media only screen and (min-width:641px) {
  .c10 {
    margin-bottom: 30px;
  }
}

@media only screen and (min-width:641px) {
  .c2 {
    margin-bottom: 0;
  }
}

<form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
  >
    <h2
      class="c0"
    >
      Access public health data
    </h2>
    <div
      class="c1"
    >
      <p>
        A free government service that provides access to a wide range of public health data in England.
      </p>
    </div>
    <br />
    <div
      class="c1 c2"
    >
      <p>
        Contents
      </p>
    </div>
    <ul
      class="c3 c4"
    >
      <li
        class="c5 c6"
      >
        <a
          class="c7"
          href="#search"
        >
          Find public health data
        </a>
      </li>
      <li
        class="c5 c6"
      >
        <a
          class="c7"
          href="#whatfor"
        >
          What the service is for
        </a>
      </li>
      <li
        class="c5 c6"
      >
        <a
          class="c7"
          href="#indicators"
        >
          About indicators
        </a>
      </li>
      <li
        class="c5 c6"
      >
        <a
          class="c7"
          href="#whofor"
        >
          Who the service is for
        </a>
      </li>
    </ul>
    <br />
    <div
      id="search"
    >
      <div
        data-testid="search-form"
      >
        <h3
          class="c8"
        >
          Find public health data
        </h3>
        <input
          hidden=""
          name="searchState"
        />
        <label
          class="c9 c10"
          data-testid="indicator-search-form-input"
        >
          <span
            class="c11"
          >
            Search by subject
          </span>
          <span
            class="c12"
          >
            <div
              style="color: rgb(80, 90, 95);"
            >
              For example, smoking, diabetes prevalence, or a specific indicator ID
            </div>
          </span>
          <input
            class="c13"
            id="indicator"
            name="indicator"
            value=""
          />
        </label>
        <div
          class="c14"
        >
          <div
            class="c15"
            data-testid="area-search-input-field"
          >
            <label
              class="c9"
            >
              <span
                class="c11"
              >
                Search by area
              </span>
              <span
                class="c12"
              >
                <div
                  style="color: rgb(80, 90, 95);"
                >
                  For example district, county, region, NHS organisation or GP practice or code
                </div>
              </span>
              <input
                class="c13"
                id="areaSearched"
                name="areaSearched"
                value=""
              />
            </label>
          </div>
        </div>
        <details
          class="c16 c17"
        >
          <summary
            class="c18 c19"
          >
            <span
              class="c20"
            >
              Filter by area
            </span>
          </summary>
          <div
            class="c21"
          >
            <div
              data-testid="select-areas-filter-panel"
            >
              <label
                class="c9 c22"
                data-testid="area-type-selector-container"
              >
                <span
                  class="c11"
                >
                  Select an area type
                </span>
                <select
                  class="c23"
                />
              </label>
              <label
                class="c9 c22"
                data-testid="group-type-selector-container"
              >
                <span
                  class="c11"
                >
                  Select a group type
                </span>
                <select
                  class="c23"
                />
              </label>
              <label
                class="c9 c22"
              >
                <span
                  class="c11"
                >
                  Select a group
                </span>
                <select
                  class="c23"
                />
              </label>
              <div
                class="c24 c25"
              >
                <span
                  class="c11 c26"
                >
                  Select one or more areas
                </span>
              </div>
            </div>
          </div>
        </details>
        <button
          class="c27"
          data-testid="search-form-button-submit"
          style="margin-top: 25px;"
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
    <hr
      class="c28"
    />
    <h3
      class="c8"
      id="whatfor"
    >
      What the service is for
    </h3>
    <div
      class="c1"
    >
      <p>
        You can use this service to:
      </p>
    </div>
    <ul
      class="c3 c29"
    >
      <li
        class="c5 c6"
      >
        Identify and compare health trends and inequalities using data visualisations
      </li>
      <li
        class="c5 c6"
      >
        Download datasets and visualisations
      </li>
      <li
        class="c5 c6"
      >
        Access raw data via an application programming interface (API)
      </li>
    </ul>
    <div
      class="c1"
    >
      <p>
        This data comes from trusted sources such as the NHS and Office of National Statistics (ONS). It is analysed to create a wide range of health and wellbeing data sets known as indicators, and themed topics known as profiles.
      </p>
    </div>
    <br />
    <h3
      class="c8"
      id="indicators"
    >
      About indicators
    </h3>
    <div
      class="c1"
    >
      <p>
        Indicators use data to show us how things are going and if there are any changes over time. For example, if the number of people smoking in South West England has gone up or down in the last ten years. These indicators can be used to compare public heath by areas, and see how it is impacted by inequalities such as age, sex or ethnicity.
      </p>
    </div>
    <br />
    <h3
      class="c8"
      id="whofor"
    >
      Who the service is for
    </h3>
    <div
      class="c1"
    >
      <p>
        The service is free and available to everyone. Its primary role is to support health professionals, policymakers, and researchers to monitor health trends, identify inequalities, and make informed decisions about public health in England.
      </p>
    </div>
    <br />
  </form>
</DocumentFragment>
`;
